mayastor/src/bdev/nexus/nexus_child.rs

```diff
@@ -266,10 +272,18 @@ impl NexusChild {
         }
         self.open(parent_size).map(|s| {
             self.status_reasons.offline(false);
+            NexusChild::save_state_change();
```

Comment:
I wonder if this is "too early", because right after we online it we need to mark it as degraded so it can be rebuilt? Eg, if we crashed in between we'd think that this child was healthy?

Reply:
Done. I've moved the setting of the out-of-sync flag to where we change the offline flag to false, so it is more of an atomic operation and prevents things like this.